File: db/src/main/java/com/delivery/db/storeuser/StoreUserRepository.java
package com.delivery.db.storeuser;

import com.delivery.db.storeuser.enums.StoreUserStatus;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface StoreUserRepository extends JpaRepository<StoreUserEntity, Long> {

    Optional<StoreUserEntity> findFirstByEmailAndStatusOrderByIdDesc(String email,
        StoreUserStatus status);

}

File: db/src/main/kotlin/com/delivery/db/storeuser/StoreUserRepository.kt
package com.delivery.db.storeuser

import com.delivery.db.storeuser.enums.StoreUserStatus
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface StoreUserRepository : JpaRepository<StoreUserEntity, Long?> {
    fun findFirstByEmailAndStatusOrderByIdDesc(
        email: String?,
        status: StoreUserStatus?
    ): StoreUserEntity?
}
